src/ducks/mobile/push.js

```diff
@@ -1,10 +1,20 @@
-/* global PushNotification, cozy */
+/* global cozy */
 import { hashHistory } from 'react-router'
+import { getSettingsFromState, isNotificationEnabled } from 'ducks/settings'
 
 let push
 
 export const registerPushNotifications = () => async (dispatch, getState) => {
-  if (push) {
+  const state = getState()
+
+  const settings = getSettingsFromState(state)
+  const client = state.mobile.client
+
+  return startPushNotifications(settings, client)
+}
+
+export const startPushNotifications = (settings, client) => {
+  if (push || !isNotificationEnabled(settings)) {
```

Comment:
I would put the `isNotificationEnabled` outside of this function. This way, this function does not need the `settings`.